pinot-segment-local/src/main/java/org/apache/pinot/segment/local/startree/v2/builder/StarTreeIndexSeparator.java

```diff
@@ -143,6 +143,7 @@ private void writeIndexToFile(File destFile, StarTreeIndexMapUtils.IndexValue va
       throws IOException {
     try (FileChannel dest = new RandomAccessFile(destFile, "rw").getChannel()) {
       org.apache.pinot.common.utils.FileUtils.transferBytes(_indexFileChannel, value._offset, value._size, dest);
+      _indexFileChannel.force(true);
```

Comment:
`_indexFileChannel` is read only. Do you intend to force `dest` here?

Reply:
Done thanks!